File: backend/src/main/java/com/yagubogu/stat/dto/AverageStatisticResponse.java
package com.yagubogu.stat.dto;

import java.math.BigDecimal;
import java.math.RoundingMode;

public record AverageStatisticResponse(
        Double averageRun,
        Double concededRuns,
        Double averageErrors,
        Double averageHits,
        Double concededHits
) {

    public static AverageStatisticResponse from(final AverageStatistic averageStatistic) {
        return new AverageStatisticResponse(
                round(averageStatistic.averageRuns()),
                round(averageStatistic.averageConcededRuns()),
                round(averageStatistic.averageErrors()),
                round(averageStatistic.averageHits()),
                round(averageStatistic.averageConcededHits())
        );
    }

    private static Double round(final Double value) {
        if (value == null) {
            return null;
        }

        return BigDecimal.valueOf(value)
                .setScale(1, RoundingMode.HALF_UP)
                .doubleValue();
    }
}

File: android/app/src/main/java/com/yagubogu/data/dto/response/AverageStatisticResponse.kt
package com.yagubogu.data.dto.response

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AverageStatisticResponse(
    @SerialName("averageRun")
    val averageRun: Double?, // 평균 득점
    @SerialName("concededRuns")
    val concededRuns: Double?, // 평균 실점
    @SerialName("averageErrors")
    val averageErrors: Double?, // 평균 실책
    @SerialName("averageHits")
    val averageHits: Double?, // 평균 안타
    @SerialName("concededHits")
    val concededHits: Double?, // 평균 피안타
)
